pages/api/gifts/[id].ts

```diff
@@ -40,24 +42,26 @@ export default async function handler(
   }
 }
 
-async function handleGET(req: NextApiRequest, res: NextApiResponse) {
-  const giftRequest = await axios.get(`${baseURL}/${queryID}`);
+async function handleGET({ res, queryId }: HandlerParams) {
+  const giftRequest = await axios.get(`${baseURL}/${queryId}`);
   return res.status(giftRequest.status).send(giftRequest.data as Gift);
 }
 
-async function handlePATCH(req: NextApiRequest, res: NextApiResponse) {
-  const patchRequest = await axios.patch(`${baseURL}/${queryID}`, req.body);
-  return res.status(patchRequest.status).json(req.body);
+async function handlePATCH({ req, res, queryId }: HandlerParams) {
+  const patchRequest = await axios.patch(`${baseURL}/${queryId}`, req.body);
+  return res.status(patchRequest.status).json(patchRequest.data as Gift);
 }
 
-async function handlePUT(req: NextApiRequest, res: NextApiResponse) {
-  const putRequest = await axios.put(`${baseURL}/${queryID}`, req.body);
-  return res.status(putRequest.status).json(req.body);
+async function handlePUT({ req, res, queryId }: HandlerParams) {
+  const putRequest = await axios.put(`${baseURL}/${queryId}`, req.body);
+  return res.status(putRequest.status).json(putRequest.data as Gift);
 }
 
-async function handleDELETE(req: NextApiRequest, res: NextApiResponse) {
-  const deleteRequest = await axios.delete(`${baseURL}/${queryID}`);
-  return res.status(deleteRequest.status).json(req.body);
+async function handleDELETE({ res, queryId }: HandlerParams) {
+  const deleteRequest = await axios.delete(`${baseURL}/${queryId}`);
+  return res
+    .status(deleteRequest.status)
+    .send('Lahja poistettu onnistuneesti!');
```

Comment:
Ei palauteta DELETE-requestista bodyä.
```suggestion
  res.status(deleteRequest.status).end();
  return;
```